### PR TITLE
UI revamp types change

### PR DESCRIPTION
**Pull request by igorRoog**
Types updated, needed to backstage-plugin UI revamp. (https://github.com/cortexapps/backstage-plugin/pull/151)

### CODE REVIEW

File: src/extensionApi.ts [hunk 1]
```diff
@@ -191,19 +203,79 @@ export interface Group {
...
+}
+
+export type ScorecardEntityFilter =
+  | CqlFilter
+  | DomainFilter
+  | ServiceFilter
+  | ResourceFilter
+  | TeamFilter;
+
 export interface Scorecard {
```
Comment (cristinabuenahora) [inline src/extensionApi.ts:267]: Do we need to add the rule exemption types here?

**Review by cristinabuenahora — APPROVED** [review]

File: src/extensionApi.ts (hunk 1 above)
Comment (igorRoog) [reply src/extensionApi.ts:267]: I don't know. There is not rule exemptions in Scorecard type in `backstage-plugin`.
And here it is all about UiExtensions and defining how Scorecard looks:

```
 scorecards?: {
        /**
         * Override default sort order of Scorecards in both the global and entity list view of Scorecards.
         */
        sortOrder?: {
            compareFn: (a: Scorecard, b: Scorecard) => number;
        };
        /**
         * Add badges to the list view of Scorecards.
         */
        cardDisplayOptions?: {
            getBadgesFn?: (scorecard: Scorecard) => string[];
        };
    };
```

On the other hand, I can safely remove `filter` field from `Scorecard` type, so it won't be visible outside and then I can remove all those new definitions. What do you think?

File: src/extensionApi.ts [hunk 2]
```diff
@@ -191,19 +203,79 @@ export interface Group {
...
   creator: {
     name: string;
     email: string;
   };
+  description?: string;
```
Comment (cristinabuenahora) [inline src/extensionApi.ts:272]: Should this be: 

```suggestion
  description: string | null;
```

File: src/extensionApi.ts (hunk 1 above)
Comment (cristinabuenahora) [reply src/extensionApi.ts:267]: I think we should have this match the Backstage Scorecard type so we should keep filter + we should add a new interface for Rule Exemptions.

File: src/extensionApi.ts (hunk 2 above)
Comment (igorRoog) [reply src/extensionApi.ts:272]: I can't do this as it won't be inline with type defined in backstage-plugin (already spotted by @FoodProduct there). We agreed not to fix that in ui-revamp PR as that PR is already quite big. (https://github.com/cortexapps/backstage-plugin/pull/152#discussion_r1470175300)

Comment (cristinabuenahora) [reply src/extensionApi.ts:272]: Makes sense

**Pull request merged**